protocol/vm/vm.go

```diff
@@ -258,3 +263,34 @@ func stackCost(stack [][]byte) int64 {
 	}
 	return result
 }
+
+type VMErr struct {
```

Comment:
Do we want to export this type? If so, it could use some documentation describing how the caller is meant to use it (e.g. https://godoc.org/os#PathError). If we don't think anyone is likely to do a `err.(vm.VMErr)` type assertion, then this might be better left unexported.

Btw, the naming convention for error types is `FooError` or `fooError`. In this case, since the package is already named `vm` it should probably be called just `Error`.